engine.py

```diff
@@ -226,6 +228,60 @@ def context_change_allowed(self):
         """
         return True
 
+    @property
+    def host_info(self):
+        """
+        :returns: A dictionary with information about the application hosting this engine.
+
+        The returned dictionary is of the following form on success:
+
+            {
+                "name": "Maya",
+                "version": "2017 Update 4",
+            }
+
+        The returned dictionary is of following form on an error preventing
+        the version identification.
+
+            {
+                "name": "Maya",
+                "version: "unknown"
+            }
+        """
+
+        host_info = {"name": "Maya", "version": "unknown"}
+        try:
+
+            # The 'about -installedVersion' Maya MEL command returns:
+            # - the app name (Maya, Maya LT, Maya IO)
+            # - the major version (2017, 2018)
+            # - the update version when applicable (update 4)
+            maya_installed_version_string = cmds.about(installedVersion=True)
```

Comment:
This looks very complicated... Would it be possible to extract everything with a single regexp matching? What are we expecting? Something like "Autodesk Maya IO 2017 Update 4"? So would the regexp below cover that:
`m=re.match("(?:Autodesk)\s+([a-zA-Z ]+)\s+(\d+)(?:\s+(\w+\s+\d+))?$", "Autodesk Maya IO 2016 Extension 2")`
- Optional "Autodesk" followed by one or more spaces.
- Letters or spaces (no digits) to extract "Maya IO".
- A bunch of digits to extract the version number.
- Optionally followed by a word and some digits, e.g. "Update 4" or "Extension 2"

Reply:
As a side note, do we want to return a semantic versioning version string, so `2017.4` by dropping the "Update" or "Extension" word? Do we enforce this convention for other host applications? (Nuke, Mari, etc...)

Reply:
I noticed that there is a "Maya 2018.1 Update" version available (same for 2017) so we need to make sure our version extraction works in that case.

Reply:
"2018.1" not "2018 Update 1" ?